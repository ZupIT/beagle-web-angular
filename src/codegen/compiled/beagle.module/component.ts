/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Type } from '@angular/core'
import {
  kebabToCamelCase,
  removeExtraIndentation,
  replaceToUnderline,
} from '../../utils/formatting'
import { getComponentAnnotations } from '../../utils/metadata'
import { remoteViewSelector } from '../../../constants'


function createViewChildString(name: string, templateName: string, angularVersion: number): string {
  const componentName = replaceToUnderline(name)

  if (angularVersion >= 8) {
    return `'${componentName}': new ViewChild('${templateName}', { static: true })`
  } else {
    return `'${componentName}': new ViewChild('${templateName}')`
  }
}

function createQueries(components: Record<string, Type<any>>, angularVersion: number) {
  const componentNames = Object.keys(components)

  return componentNames.map((name) => {
    const selector = getComponentAnnotations(components[name]).selector
    const templateName = kebabToCamelCase(selector)
    return createViewChildString(name, templateName, angularVersion)
  })
}


export function createComponentString(
    components: Record<string, Type<any>>,
    angularVersion: number,
  ) {
  const queries = createQueries(components, angularVersion)

  // todo: legacy code. Remove the input "loadParams" with v2.0.
  const componentString = `
    @Component({
      selector: '${remoteViewSelector}',
      template,
      inputs: ['loadParams', 'route', 'networkOptions', 'controllerId'],
      queries: {
        ${queries.join(',\n        ')},
      },
    })
    export class BeagleRemoteView extends AbstractBeagleRemoteView {
      constructor(
        beagleProvider: BeagleProvider,
        ngZone: NgZone,
        changeDetector: ChangeDetectorRef,
      ) {
        // @ts-ignore
        super(beagleProvider, ngZone, changeDetector)
      }
    }
  `

  return removeExtraIndentation(componentString, 4)
}
