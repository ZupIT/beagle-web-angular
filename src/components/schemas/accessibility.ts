export interface Accessibility {
  accessible: boolean,
  accessibilityLabel?: string,
  isHeader?: boolean,
}
