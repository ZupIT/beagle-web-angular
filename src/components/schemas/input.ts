/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Properties as CSSProperties } from 'csstype'
import { InputHandler } from './input-handler'

export type InputType = 'DATE' | 'EMAIL' | 'PASSWORD' | 'NUMBER' | 'TEXT'

export interface BeagleInputInterface {
  value?: string,
  placeholder?: string,
  disabled?: boolean,
  readOnly?: boolean,
  type?: InputType,
  hidden?: boolean,
  styleId?: string,
  style?: CSSProperties,
  onChange?: InputHandler,
  onBlur?: InputHandler,
  onFocus?: InputHandler,
  error?: string,
  showError?: boolean,
}
