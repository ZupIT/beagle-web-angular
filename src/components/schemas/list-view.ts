/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { BeagleUIElement } from '@zup-it/beagle-web'
import { Properties as CSSProperties } from 'csstype'

export type Direction = 'VERTICAL' | 'HORIZONTAL'

export interface BeagleFutureListViewInterface {
  direction: Direction,
  dataSource: any[],
  iteratorName?: string,
  onInit?: () => void,
  onScrollEnd?: () => void,
  scrollEndThreshold?: number,
  template: BeagleUIElement,
  useParentScroll?: boolean,
  style?: CSSProperties,
}

export interface BeagleListViewInterface {
  direction: Direction,
}
