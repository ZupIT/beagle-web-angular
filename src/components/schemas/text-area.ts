/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Properties as CSSProperties } from 'csstype'

export type InputHandler = (event: { value: string }) => void

export interface TextAreaInterface {
  value?: string,
  label?: string,
  name?: string,
  onChange?: InputHandler,
  onFocus?: InputHandler,
  onBlur?: InputHandler,
  disabled?: boolean,
  readonly?: boolean,
  styleId?: string,
  style?: CSSProperties,
}
