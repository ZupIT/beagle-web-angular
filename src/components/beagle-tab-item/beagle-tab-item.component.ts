/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { Component, Input, OnInit, ViewEncapsulation } from '@angular/core'
import { BeagleComponent } from '../../runtime/BeagleComponent'
import { TabsService } from '../services/tabs.service'
import { BeagleTabItemInterface } from '../schemas/tab-view'

@Component({
  selector: 'beagle-tab-item',
  templateUrl: './beagle-tab-item.component.html',
  styleUrls: ['./beagle-tab-item.component.less'],
  encapsulation: ViewEncapsulation.None,
})
export class BeagleTabItemComponent extends BeagleComponent
  implements BeagleTabItemInterface, OnInit {

  @Input() title?: string
  @Input() icon?: string
  active = false
  id: string

  constructor(private tabsService: TabsService) {
    super()
  }

  ngOnInit() {
    this.id = this.getBeagleContext().getElementId()
    this.listenTabChanges()
  }

  listenTabChanges() {
    this.tabsService.notifySelectedTab().subscribe((selected) => {
      this.active = selected === this.id
    })
  }

  handleClick() {
    const item = this.getBeagleContext().getElementId()
    this.tabsService.changeSelectedTab(item)
  }
}
