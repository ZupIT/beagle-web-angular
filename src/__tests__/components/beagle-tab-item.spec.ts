/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import { TestBed, async, fakeAsync, ComponentFixture } from '@angular/core/testing'
import { BeagleTabItemComponent } from '../../components/beagle-tab-item/beagle-tab-item.component'
import { TabsService } from '../../components/services/tabs.service'
import { ImagePath, ImagePathMode } from '../../components/schemas/image'
import mockBeagleContext, { setAndCallHandler } from './mocks/test-mocks.spec'

let component: BeagleTabItemComponent
let fixture: ComponentFixture<any>
const imagePathModeMock: ImagePathMode = 'local'
const pathMock: ImagePath = {
  _beagleImagePath_: imagePathModeMock,
  url: 'http://teste.com.br',
}
let tabServiceMock: TabsService

describe('BeagleTabItemComponent', () => {


  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        BeagleTabItemComponent,
      ],
      providers: [
        TabsService,
      ],
    }).compileComponents()
    fixture = TestBed.createComponent(BeagleTabItemComponent)
    tabServiceMock = TestBed.get(TabsService)
    spyOn(tabServiceMock, 'notifySelectedTab').and.callThrough()
    spyOn(tabServiceMock, 'changeSelectedTab').and.callThrough()
    component = fixture.componentInstance
    component.getViewContentManager = () => mockBeagleContext
  }))

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('should retrieve component id call listenTabChanges', () => {
    spyOn(component, 'ngAfterViewInit').and.callThrough()
    spyOn(component, 'listenTabChanges').and.callThrough()
    component.ngAfterViewInit()
    expect(component['id']).toBe('abcd')
    expect(component.listenTabChanges).toHaveBeenCalled()
    expect(tabServiceMock.notifySelectedTab).toHaveBeenCalled()
  })

  it('should call on click Handler', fakeAsync(() => {
    spyOn(component, 'handleClick').and.callThrough()

    setAndCallHandler('div', 'Testing', 'click', fixture)

    expect(component.handleClick).toHaveBeenCalled()
  }))

})
