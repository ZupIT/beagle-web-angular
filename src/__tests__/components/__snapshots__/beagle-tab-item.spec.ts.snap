// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeagleTabItemComponent should match snapshot 1`] = `
BeagleTabItemComponent {
  "active": false,
  "getViewContentManager": [Function],
  "tabsService": TabsService {
    "changeSelectedTab": [Function],
    "notifySelectedTab": [Function],
    "selectedTab": BehaviorSubject {
      "_isScalar": false,
      "_value": "",
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
  },
}
`;
